app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/Matchers.ts

```diff
@@ -22,64 +32,120 @@ export class Matchers {
-      this.testCase.errors.push(
-        defaultTestErrorLog(expectedValue, this.currentValue, comparisonMode)
-      );
-    }
-  }
 
-  public toBeCalled(times = 1) {
-    this.assertValueIsCallTracker(this.currentValue);
-    const callsCount = this.currentValue.onUI + this.currentValue.onJS;
-    if (callsCount !== times) {
-      const name = color(this.currentValue.name, 'green');
-      const expected = color(times, 'green');
-      const received = color(callsCount, 'red');
-      this.testCase.errors.push(
-        `Expected ${name} to be called ${expected} times, but was called ${received} times`
-      );
-    }
-  }
+    const coloredExpected = color(expectedValue, 'green');
+    const coloredReceived = color(currentValue, 'red');
+    const coloredMode = color(comparisonMode, 'yellow');
+
+    return {
+      pass: isEqual(expectedValue, currentValue),
+      message: `Expected${
+        this._negation ? ' NOT' : ''
+      } ${coloredExpected} received ${coloredReceived}, mode: ${coloredMode}`,
+    };
+  };
+
+  private _toBeCalledMatcher: MatcherFunction = (
+    currentValue: TestValue,
+    times = 1
+  ) => {
+    Matchers._assertValueIsCallTracker(currentValue);
+    const callsCount = currentValue.onUI + currentValue.onJS;
+    const name = color(currentValue.name, 'green');
+    const expected = color(times, 'green');
+    const received = color(callsCount, 'red');
+    return {
+      pass: callsCount === times,
+      message: `Expected ${name}${
+        this._negation ? ' NOT' : ''
+      } to be called ${expected} times, but was called ${received} times`,
+    };
+  };
 
-  public toBeCalledUI(times = 1) {
-    this.assertValueIsCallTracker(this.currentValue);
-    if (this.currentValue.onUI !== times) {
-      const name = color(this.currentValue.name, 'green');
-      const threadName = color('UI thread', 'cyan');
-      const expected = color(times, 'green');
-      const received = color(this.currentValue.onUI, 'red');
-      this.testCase.errors.push(
-        `Expected ${name} to be called ${expected} times on ${threadName}, but was called ${received} times`
+  private _toBeCalledUIMatcher: MatcherFunction = (
+    currentValue: TestValue,
+    times = 1
+  ) => {
+    Matchers._assertValueIsCallTracker(currentValue);
+    const callsCount = currentValue.onUI;
+    const name = color(currentValue.name, 'green');
+    const threadName = color('UI thread', 'cyan');
+    const expected = color(times, 'green');
+    const received = color(callsCount, 'red');
+
+    return {
+      pass: callsCount === times,
+      message: `Expected ${name}${
+        this._negation ? ' NOT' : ''
+      } to be called ${expected} times on ${threadName}, but was called ${received} times`,
+    };
+  };
+
+  private _toBeCalledJSMatcher: MatcherFunction = (
+    currentValue: TestValue,
+    times = 1
+  ) => {
+    Matchers._assertValueIsCallTracker(currentValue);
+    Matchers._assertValueIsCallTracker(currentValue);
```

Comment:
We can call it only once
```suggestion
    Matchers._assertValueIsCallTracker(currentValue);
```